ush/biasCorrectionReader.py

```diff
@@ -0,0 +1,312 @@
+import yaml
+from netCDF4 import Dataset
+import numpy as np
+import netCDF4 as nc
+from datetime import datetime
+from eva.utilities.logger import Logger
+
+
+def read_ncfile(file, groups, variable, channels, logger):
+    """
+    Read and extract data from ncfile based on inputted groups.
+
+    Parameters:
+        file (str): path to input netCDF4 file
+        group_list (list): list of strings representing the groups needed in
+                           nc file
+    Return:
+        d (dict): dictionary of data
+    """
+    d = {}
+
+    try:
+        with Dataset(file, mode='r') as f:
+            # Grab index of input channels
+            chan_numbers = f.groups['MetaData'].variables['sensorChannelNumber'][:][0]
+            chan_idx = np.where(np.in1d(chan_numbers, channels))[0]
+
+            for g in groups:
+                d[g] = {}
+                gd = f.groups[g]
+
+                data = gd.variables[variable][:]
+                d[g][variable] = data[:, chan_idx]
+
+    except Exception as e:
+        logger.abort("Extraction of data failed. Ensure group exists " +
+                     f"in input .ncfile. Error: {e}")
+
+    return d
+
+
+def write_ncfile(outfile, outdata, epoch_time, nchannels):
+    """
+    Write an ncfile from dictionary.
+
+    Parameters:
+        outfile (str): path and file name of where the data is to be written to specified
+                       in input yaml
+        outdata (dict): data collected from input netCDF file with calculated statistics
+        epoch_time (str): cycle time as seconds from Jan. 1, 1970
+        nchannels (int): total number of channels
+    Return:
+        None
+    """
+    # Create a new netCDF file
+    nc_file = nc.Dataset(outfile, 'w', format='NETCDF4')
+
+    valid_time_dim = nc_file.createDimension('validTime', 1)
```

Comment:
This is more question than suggestion, again out of ignorance of NetCDF.Dataset.  Is it necessary to preserve returned valid_time_dim and channel_dim?  I don't see them used later though it was helpful to have them named as they are; they make the purpose of the method clear.